insteon_mqtt/device/HiddenDoor.py

```diff
@@ -0,0 +1,652 @@
+            })
+
+        # This allows for a short timer between sending automatic battery
+        # requests.  Otherwise, a request may get queued multiple times
+        self._battery_request_time = 0
+
+    #-----------------------------------------------------------------------
+    @property
+    def battery_voltage_time(self):
+        """Returns the timestamp of the last battery voltage report from the
+        saved metadata
+        """
+        meta = self.db.get_meta('HDS')
+        ret = 0
+        if isinstance(meta, dict) and 'battery_voltage_time' in meta:
+            ret = meta['battery_voltage_time']
+        return ret
+
+    #-----------------------------------------------------------------------
+    @battery_voltage_time.setter
+    def battery_voltage_time(self, val):
+        """Saves the timestamp of the last battery voltage report to the
+        database metadata
+        Args:
+          val:    (timestamp) time.time() value
+        """
+        meta = {'battery_voltage_time': val}
+        existing = self.db.get_meta('HDS')
+        if isinstance(existing, dict):
+            existing.update(meta)
+            self.db.set_meta('HDS', existing)
+        else:
+            self.db.set_meta('HDS', meta)
+
+    #-----------------------------------------------------------------------
+    @property
+    def battery_low_voltage(self):
+        """Returns the voltage below which the battery will be deemed to be
+        low.  The default value is 1.25V
+        """
+        meta = self.db.get_meta('HDS')
+        ret = 1.25
+
+        if isinstance(meta, dict) and 'battery_low_voltage' in meta:
+            ret = meta['battery_low_voltage']
+        return ret
+
+    #-----------------------------------------------------------------------
+    @battery_low_voltage.setter
+    def battery_low_voltage(self, val):
+        """Saves the voltage below which the battery will be deemed to be
+        low.
+        Args:
+          val:    (float) Low voltage number
+        """
+        meta = {'battery_low_voltage': val}
+        existing = self.db.get_meta('HDS')
+        if isinstance(existing, dict):
+            existing.update(meta)
+            self.db.set_meta('HDS', existing)
+        else:
+            self.db.set_meta('HDS', meta)
+
+    #-----------------------------------------------------------------------
+    def set_low_battery_voltage(self, on_done, voltage=None):
+        """Set low voltage value.
+
+        Called from the mqtt command functions or cmd_line
+
+        Args:
+          voltage: (int) The low voltage value
+          on_done: Finished callback.  This is called when the command has
+                   completed.  Signature is: on_done(success, msg, data)
+        """
+        if voltage is not None:
+            LOG.info("Hidden Door %s cmd: set low voltage= %s", self.label,
+                     voltage)
+            self.battery_low_voltage = voltage
```

Comment:
It may be better to wait until we get the ack before saving the new value.

Reply:
I copied this from the motion sensor doing the same thing.  @krkeegan - thoughts?